libimage/layer_tree.go

```diff
@@ -136,14 +136,23 @@ func (r *Runtime) newLayerTreeFromData(ctx context.Context, images []*Image, lay
 			if !generateManifestDigestList {
 				continue
 			}
-			if manifestList, _ := img.IsManifestList(ctx); manifestList {
-				mlist, err := img.ToManifestList()
-				if err != nil {
-					return nil, err
-				}
-				for _, digest := range mlist.list.Instances() {
-					tree.manifestListDigests[digest] = struct{}{}
+			// ignore errors, common errors are
+			//  - image is not manifest
+			//  - image has been removed from the store in the meantime
+			// In all cases we should ensure image listing still works and not error out.
+			mlist, err := img.ToManifestList()
+			if err != nil {
+				// If it is not a manifest it likely is a regular image so just ignore it.
+				// If the image is unknown that likely means there was a race where the image/manifest
+				// was removed after out MultiList() call so we ignore that as well.
+				if errors.Is(err, ErrNotAManifestList) || errors.Is(err, storageTypes.ErrImageUnknown) {
+					continue
 				}
+				return nil, err
```

Comment:
Consider wrapping this with the image ID.

Reply:
added wrapping in LoadFromImage()